src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateSet.cs

```diff
@@ -265,19 +256,34 @@ public void OnRemoved()
                 InMemoryStorage.DropCache(Analyzer);
             }
 
-            /// <summary>
-            /// Get the unique state name for the given analyzer.
-            /// Note that this name is used by the underlying persistence stream of the corresponding <see cref="ProjectState"/> to Read/Write diagnostic data into the stream.
-            /// If any two distinct analyzer have the same diagnostic state name, we will end up sharing the persistence stream between them, leading to duplicate/missing/incorrect diagnostic data.
-            /// </summary>
-            private static ValueTuple<string, VersionStamp> GetNameAndVersion(DiagnosticAnalyzer analyzer)
+            private sealed class AnalyzerTypeData
             {
-                Contract.ThrowIfNull(analyzer);
+                private const string UserDiagnosticsPrefixTableName = "<UserDiagnostics2>";
+                private static readonly ConditionalWeakTable<Type, AnalyzerTypeData> s_analyzerTypeData
```

Comment:
Why bother with `ConditionalWeakTable` here - we never unload analyzers, so the `Type` is never going to go away.

Reply:
It's the API that felt most natural to me for this. It could be switched out for an alternative.

Reply:
I would just make it concurrentDictionary<string /* analyzerId */, AnalyzerTyepData>. and don't care about remove analyzerId from it.

simpler, and even thought it can leak, in practice won't matter much. we already has such cache in many places.

Reply:
➡️ Fixed in 6200510